### ISSUE TITLE
Suggestion for named fields in sum types

### ISSUE THREAD

**Issue by sjakobi** [ndmitchell/hlint#217]
The problem with types like
```
data Foo
  = Bar {bar :: Baz}
  | Blup
```
is that they introduce partial functions (`bar` in this case). The recommendation is therefore to use named fields only in true product types.

**Comment by ndmitchell** [comment]
I can see why GHC might want this as a warning, but there's no easy refactoring to avoid it, and record systems are a place where everyone has their own opinion, so my inclination is that HLint shouldn't give a warning that can't be easily remedied.

**Comment by sjakobi** [comment]
Fair point, I'll open a GHC-ticket when I get around to it.
